lib/view_component/base.rb

```diff
@@ -75,7 +75,7 @@ def render_in(view_context, &block)
       before_render
 
       if render?
-        send(self.class.call_method_name(@variant))
+        send(self.class.call_method_name(@variant)).gsub(/>[\n\s]+</, "><").strip.html_safe
```

Comment:
I'm not sure if this is the right place or we should extract it?

Reply:
This is probably fine. Not being a _regex-pert_ myself, do you think we need any more test cases for this replacement pattern?

Reply:
This should remove spaces and newlines. which is covered by this test https://github.com/github/view_component/pull/416/files#diff-c7432794f4e04debce14f00887227594R648-R654 (the test partial has both)